File: views/find_missing/map.js
function(doc) {
  if (doc.missing && doc.missing.length > 0) {
      var failed = doc.failed.map(function(n) {
          return parseInt(n.id);
      }).sort(function (a, b) { return a - b; });
      emit([doc.test, doc.n_dead_nodes],
           {"algorithm": doc.algorithm,
            "missing": doc.missing,
            "n_missing": doc.missing.length,
            "down": doc.n_dead_nodes,
            "up": doc.n_alive_nodes,
            "failed": failed});
  }
}

File: _docs/erl/views/find_missing/map.erl
fun({Doc}) ->
        F = fun(B) -> proplists:get_value(B, Doc) end,
        FD = fun(B, D) -> proplists:get_value(B, Doc, D) end,
        case length(FD(<<"missing">>, [])) of
            0 ->
                %% Don't care
                ok;
            undefined ->
                %% Don't care
                ok;
            NMissing ->
                Failed0 = F(<<"failed">>),
                Failed = lists:map(fun ({X}) ->
                                           proplists:get_value(<<"id">>, X)
                                   end,
                                   Failed0),
                Emit([F(<<"test">>), F(<<"n_dead_nodes">>)],
                     {[
                       {<<"algorithm">>, F(<<"algorithm">>)},
                       {<<"missing">>, F(<<"missing">>)},
                       {<<"n_missing">>, NMissing},
                       {<<"down">>, F(<<"n_dead_nodes">>)},
                       {<<"up">>, F(<<"n_alive_nodes">>)},
                       {<<"failed">>, Failed}
                      ]})
        end
end.
